protocol/contracts/v2/TreasuryPriceIndexOracle.sol

```diff
@@ -46,40 +51,47 @@ contract TreasuryPriceIndexOracle is ITreasuryPriceIndexOracle, TempleElevatedAc
      */
     uint256 public override maxTreasuryPriceIndexDelta;
 
+    /**
+     * @notice The maximum allowed TPI change on any single `setTreasuryPriceIndex()`, in absolute terms.
+     * @dev Used as a bound to avoid unintended/fat fingering when updating TPI
+     */
+    uint32 public override maxTreasuryPriceIndexTargetDateDelta;
+
     constructor(
         address _initialRescuer,
         address _initialExecutor,
         uint96 _initialTreasuryPriceIndex,
         uint256 _maxTreasuryPriceIndexDelta,
-        uint32 _cooldownSecs
+        uint32 _maxTreasuryPriceIndexTargetDateDelta
     ) TempleElevatedAccess(_initialRescuer, _initialExecutor)
     {
         tpiData = TpiData({
             currentTpi: _initialTreasuryPriceIndex,
-            previousTpi: _initialTreasuryPriceIndex,
+            targetTpi: _initialTreasuryPriceIndex,
             lastUpdatedAt: uint32(block.timestamp),
-            cooldownSecs: _cooldownSecs
+            targetDate: uint32(block.timestamp),
+            tpiSlope: 1,
+            increaseInTargetTpi: true
         });
         maxTreasuryPriceIndexDelta = _maxTreasuryPriceIndexDelta;
+        maxTreasuryPriceIndexTargetDateDelta = _maxTreasuryPriceIndexTargetDateDelta;
     }
 
     /**
      * @notice The current Treasury Price Index (TPI) value
      * @dev If the TPI has just been updated, the old TPI will be used until `cooldownSecs` has elapsed
      */
     function treasuryPriceIndex() public override view returns (uint96) {
-        return (block.timestamp < (tpiData.lastUpdatedAt + tpiData.cooldownSecs))
-            ? tpiData.previousTpi  // use the previous TPI if we haven't passed the cooldown yet.
-            : tpiData.currentTpi;  // use the new TPI
-    }
-
-    /**
-     * @notice Set the number of seconds to elapse before a new TPI will take effect.
-     */
-    function setTpiCooldown(uint32 cooldownSecs) external override onlyElevatedAccess {
-        emit TpiCooldownSet(cooldownSecs);
-        tpiData.cooldownSecs = cooldownSecs;
-    }
+        if (block.timestamp >= tpiData.targetDate) {
```

Comment:
Pulling comment from Frontier on direct commit (which I may have nuked)

> Instead of a separate variable saying whether it's positive or negative, could tpiData.tpiSlope just be an int96 instead?
> 
> Would simplify just to this then?
> `return uint96(tpiData.currentTpi + (((tpiData.targetDate - tpiData.lastUpdatedAt) - (tpiData.targetDate - block.timestamp)) * tpiData.tpiSlope));`
> Any chance the second part is > tpiData.currentTpi (meaning this could underflow)? Or do we have guards against that elsewhere?

Reply:
ACK - will adjust. Previously looked at int usage across the board - went with a 'simplier' approach but agree didn't hit the mark here.

Reply:
Done in https://github.com/TempleDAO/temple/pull/1096/commits/148c89330e66ed9987bf9c5b0d9f691e329d8cf5